src/app/core/services/kafka.service.ts

```diff
@@ -70,7 +70,7 @@ export class KafkaService {
     // NOTE: Payload for kafka 1 : value Object which contains individual questionnaire response with timestamps
     const CompletionLog: CompletionLogValueExport = {
       name: task.name.toString(),
-      time: getSeconds({ milliseconds: task.timestamp }),
+      time: getSeconds({ milliseconds: new Date().getTime() + Math.random() }),
```

Comment:
WE can maybe rethink the design of this and caching. Right now it is fine to leave it as it is but later we must revisit this and think of some strategy to use for caching (like taking a hash of the timestamp) instead of using the timestamp directly as a cache key.

Reply:
Yup that's a good idea.